openhtf/core/__init__.py

```diff
@@ -50,23 +50,27 @@ class TestStopError(Exception):
 class TestExecutor(threads.KillableThread):
   """Encompasses the execution of a single test."""
 
-  def __init__(self, test_descriptor, test_start, teardown_function=None):
+  def __init__(self, test_descriptor, test_start_function, teardown_function=None):
     super(TestExecutor, self).__init__(name='TestExecutorThread')
     self.test_state = None
 
-    # Force teardown function timeout, otherwise we can hang for a long time
-    # when shutting down, such as in response to a SIGINT.
-    timeout_s = conf.teardown_timeout_s
-    if hasattr(teardown_function, 'timeout_s'):
-      timeout_s = teardown_function.timeout_s
+    # Ensure test_start is a phase that saves to the dut_id.
+    if not isinstance(test_start_function, (openhtf.TestPhase, openhtf.PhaseDescriptor)):
+      @openhtf.TestPhase()
```

Comment:
`TestPhase()` was created by @grybmadsci to make the name make more sense to others, since it clearly converts a function into a test phase, not a `PhaseOptions` instance.